### DISCUSSION TITLE
UNet 3D segmentation prediction on a single numpy array

### DISCUSSION THREAD

**Discussion by radiplab** [Project-MONAI/tutorials#425] [Q&A]
Thanks for the great tutorials - they got me up and running fast and are very helpful.

I've been using the [3D spleen segmentation tutorial](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d.ipynb). One thing I've struggled with is using a trained model to then make single predictions on images. More specifically, an already loaded numpy array of the image data (I do manipulations to the image and want to then make predictions without resaving to .nii and reloading). I suspect the "Evaluation on original image spacings" section of the tutorial is close to what I need, and perhaps using an ArrayDataset? Are there any additional examples that illustrate exactly how that could be done? It's such a common task, I want to do it the "right" way within the Monai framework. Thanks!

**Comment by Nic-Ma** [comment]
Hi @radiplab ,

Thanks for your interest here.
If you you already loaded the data into numpy array in memory, I think you just need to remove the `LoadImaged` transform, others are same as the example, right?

Thanks.
